CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/Institution.java

```diff
@@ -607,6 +613,19 @@ ProtoGrade getGrade(final String authId, final ProtoGrade gradeData)
      * @throws AuthenticationException
      *         Thrown if the user did not have the authentication to get the course.
      */
-    List<String> getCourseRoster(final String authId, final String courseId)
+    Identity.UserNameResponse getCourseRoster(final String authId, final String courseId)
+            throws DatabaseAccessException, AuthenticationException;
+
+    /**
+     * Gets the username for the userId.
+     *
+     * @param userId The userId that is being exchanged for the userId.
+     * @param authId The authentication of the user that is exchanging the userId.
+     * @param courseId The course the username is being asked for.
+     * @return The username if the permissions are successful.
+     * @throws DatabaseAccessException Thrown if the user does not exist or there are problems getting the username.
+     * @throws AuthenticationException Thrown if the user does not have the authentication to get the username.
+     */
+    String getUserNameForIdentity(final String userId, final String authId, final String courseId)
```

Comment:
In getAllAssignmentGradesStudent you put userId last
pick an order and stick to it so it is constant for all methods